### PR TITLE
flutter-web: fix access to manifest with authentication

### PR DESCRIPTION
**Pull request by PaulGrandperrin**
When the website is protected by HTTP authentication, Chrome fails to load the `manifest.json` with error 401 because it tries to access it without the authentication header.

This is surprising but apparently normal, see: https://developer.mozilla.org/en-US/docs/Web/Manifest#deploying_a_manifest

I didn't open an issue because this is fairly trivial.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter_tools/templates/app_shared/web/index.html.tmpl [hunk 1]
```diff
@@ -30,7 +30,7 @@
   <link rel="icon" type="image/png" href="favicon.png"/>
 
   <title>{{projectName}}</title>
-  <link rel="manifest" href="manifest.json">
+  <link rel="manifest" href="manifest.json" crossorigin="use-credentials">
```
Comment (ditman) [inline packages/flutter_tools/templates/app_shared/web/index.html.tmpl:33]: Wouldn't it be simpler to just comment above the `link rel="manifest"` something like:

```html
<!-- If your app is protected behind simple http auth, you need to set crossorigin="use-credentials" in the following tag. -->
```

That way we don't flip the default for a use case that is not that common?

**Review by ditman — COMMENTED** [review]
QQ for @eyebrowsoffire... Do you think setting the manifest as `crossorigin="use-credentials"` would break [cross-origin isolation](https://developer.chrome.com/blog/coep-credentialless-origin-trial)?

---

(PS: The branch needs to be updated before it can land, it seems from the user test failures?)

**Comment by Piinks** [comment]
(PR Triage): What is the status of this PR?

File: packages/flutter_tools/templates/app_shared/web/index.html.tmpl (hunk 1 above)
Comment (PaulGrandperrin) [reply packages/flutter_tools/templates/app_shared/web/index.html.tmpl:33]: I don't have an opinion, would you like me to change the PR?

**Comment by PaulGrandperrin** [comment]
It has been suggested to make those changes as comments to avoid unanticipated consequences.

I don't have a strong opinion.

File: packages/flutter_tools/templates/app_shared/web/index.html.tmpl (hunk 1 above)
Comment (chunhtai) [reply packages/flutter_tools/templates/app_shared/web/index.html.tmpl:33]: according the @ditman , I think yes

**Comment by chunhtai** [comment]
This is still not resolved https://github.com/flutter/flutter/pull/154012#discussion_r1783597776

**Comment by bkonyi** [comment]
@PaulGrandperrin just checking in. Do you plan on addressing the comment above?

**Comment by ditman** [comment]
I think we can close this, we're planning on removing the default Service Worker in this quarter, here's some documentation:

* https://groups.google.com/g/flutter-announce/c/0Vv-j_TyrdI
* https://flutter.dev/go/web-cleanup-service-worker

WDYT @bkonyi?

**Comment by bkonyi** [comment]
@ditman SGTM!

**Pull request closed without merging**